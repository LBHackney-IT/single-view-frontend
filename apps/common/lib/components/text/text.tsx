import React, { forwardRef } from "react";

import cn from "classnames";

import type * as Polymorphic from "@radix-ui/react-polymorphic";
import "./styles.scss";

export interface TextProps {
  variant?: "base" | "regular" | "bold";
  size?: "base" | "xs" | "sm" | "md" | "lg";
}

export type TextComponent = Polymorphic.ForwardRefComponent<"p", TextProps>;

export const Text: TextComponent = forwardRef(function Text(
  { as: TextComp = "p", variant = "base", size = "md", className, ...props },
  ref,
) {
  return (
    <TextComp
      ref={ref}
      className={cn(
        {
          "lbh-body-xs": size === "xs",
          "lbh-body-s": size === "sm",
          "lbh-body-m": size === "md",
          "lbh-body-l": size === "lg",
          "lbh-!-font-weight-regular": variant === "regular",
          "lbh-!-font-weight-bold": variant === "bold",
        },
        className,
      )}
      {...props}
    />
  );
});
