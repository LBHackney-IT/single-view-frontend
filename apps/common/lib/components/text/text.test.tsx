import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { Text } from "./text";

test("it renders correctly", async () => {
  const { container } = render(<Text>Hello World</Text>);
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders with variants and sizes", async () => {
  const { container } = render(
    <Text size="lg" variant="bold">
      Hello World
    </Text>,
  );
  expect(container).toMatchSnapshot();
});
