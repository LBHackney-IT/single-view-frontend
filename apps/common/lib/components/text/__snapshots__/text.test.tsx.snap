// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <p
    class="lbh-body-m"
  >
    Hello World
  </p>
</div>
`;

exports[`it renders with variants and sizes 1`] = `
<div>
  <p
    class="lbh-body-l lbh-!-font-weight-bold"
  >
    Hello World
  </p>
</div>
`;
